walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,92 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    long duration = request.getDuration();
+
+    //Case in which the requested meeting has no attendies
+    if(attendees.isEmpty()){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    //Case in which the requested meeting has a duration that is too long/short
+    if(duration > TimeRange.WHOLE_DAY.duration() || duration < 0){
+        return Arrays.asList();
+    } 
+
+    //List of time ranges of other events that attendees are attending
+    List<TimeRange> timeRanges = new ArrayList<TimeRange>();
+    TimeRange currentRange;
+    for(String person : attendees){
+        for(Event event: events){
+            currentRange = event.getWhen();
+            if(event.getAttendees().contains(person) && !timeRanges.contains(currentRange)){
+                timeRanges.add(currentRange);
+            }
+        }
+    }
```

Comment:
Consider extracting this part as a helper method for better readability. Also practicing java stream here might be a good idea.